### PR TITLE
Kk/remove unused methods

### PR DESCRIPTION
**Pull request by krzkaczor**
- remove Insert & Bump

### CODE REVIEW

**Comment by krzkaczor** [comment]
@maciejka to remove unordered list entirely we would need to kill off bigger part of the SC:
- matchingEnabled switch
- methods: `_offeru`, `_sort`, `_unsort`, `getFirstUnsortedOffer`, `getNextUnsortedOffer`, `setMatchingEnabled`
- events: `LogMatchingEnabled`
- tweak logic in: `offer`, `buy`, `cancel`, 

Do we want to do broad changes?

**Comment by krzkaczor** [comment]
I removed more code but I am not so sure about these changes.... @maciejka

File: src/matching_market.sol [hunk 1]
```diff
@@ -106,8 +100,7 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
         returns (uint)
     {
         require(!locked, "Reentrancy attempt");
-        var fn = matchingEnabled ? _offeru : super.offer;
-        return fn(pay_amt, pay_gem, buy_amt, buy_gem);
+        return super.offer(pay_amt, pay_gem, buy_amt, buy_gem);
```
Comment (krzkaczor) [inline src/matching_market.sol]: should this be a noop? reverse? should we delete matchingEnabled flag? note that `_offeru` called super.offer anyway...

Comment (gbalabasquer) [reply src/matching_market.sol]: Hmmm regarding this function, I think we could just remove it at all. As you said it doesn't do anything excepting checking for the reentrancy attempt. However the original `offer` function in `simple_market` has the `synchronized` modifier that already checks it. So yes IMO we should just remove the function.

Regarding `matchingEnabled`, at this point I'd say we also remove all regarding this. If the idea is making the contract simpler is a good step.

The only thing I'm concerned is we are working in this old version which doesn't even allow to compile with solc 0.5.

**Pull request closed without merging**